src/crimson/osd/pg.cc

```diff
@@ -490,18 +491,29 @@ seastar::future<> PG::submit_transaction(ObjectContextRef&& obc,
 					 const MOSDOp& req)
 {
   epoch_t map_epoch = get_osdmap_epoch();
-  eversion_t at_version{map_epoch, projected_last_update.version + 1};
+  eversion_t at_version{map_epoch, ++projected_last_update.version};
+
+  vector<pg_log_entry_t> logv;
+  logv.push_back(pg_log_entry_t(obc->obs.exists ? pg_log_entry_t::MODIFY : pg_log_entry_t::DELETE,
```

Comment:
nit, could use `emplace_back()`, to 1) construct the `pg_log_entry_t` in-place, 2) avoid repeating `pg_log_entry_t`.